estate/models/estate_property_offer.py

```diff
@@ -15,3 +18,33 @@ class EstatePropertyOffer(models.Model):
     )
     partner_id = fields.Many2one("res.partner", required=True)
     property_id = fields.Many2one("estate.property", required=True)
+    validity = fields.Integer(default=7, string="Validity (days)")
+    date_deadline = fields.Date(
+        compute="_compute_date_deadline",
+        inverse="_inverse_date_deadline",
+        string="Deadline",
+    )
+
+    @api.depends("create_date", "validity")
+    def _compute_date_deadline(self):
+        for record in self:
+            record.date_deadline = record.create_date + relativedelta(days=record.validity)
+
+    def _inverse_date_deadline(self):
+        for record in self:
+            record.validity = (record.date_deadline - record.create_date.date()).days
+
+    def action_accept_offer(self):
+        for record in self:
+            if "accepted" in record.property_id.offer_ids.mapped("status"):
+                raise UserError(_("Another offer have been accepted"))
+
+            record.status = "accepted"
+            record.property_id.buyer_id = record.partner_id
+            record.property_id.selling_price = record.price
+        return True
+
+    def action_refuse_offer(self):
+        for record in self:
+            record.status = "refused"
```

Comment:
You could avoid the for, using the [write](https://www.odoo.com/documentation/16.0/developer/reference/backend/orm.html#odoo.models.Model.write) method. Please read about it.

```suggestion
        self.write({'status': "refused"})
```